src/ProjectTask.php

```diff
@@ -583,315 +584,45 @@ public static function getAllTicketsForProject($ID)
-        }
-
-        Dropdown::showTimeStamp(
-            "planned_duration",
-            ['min'             => 0,
-                'max'             => 8 * HOUR_TIMESTAMP,
-                'rand'            => $rand_planned_duration,
-                'value'           => $this->fields["planned_duration"],
-                'addfirstminutes' => true,
-                'inhours'         => true,
-                'toadd'           => $toadd
-            ]
-        );
-        echo "</td>";
-        echo "<td>" . __('Effective duration') . "</td>";
-        echo "<td>";
-        Dropdown::showTimeStamp(
-            "effective_duration",
-            ['min'             => 0,
-                'max'             => 8 * HOUR_TIMESTAMP,
-                'rand'            => $rand_effective_duration,
-                'value'           => $this->fields["effective_duration"],
-                'addfirstminutes' => true,
-                'inhours'         => true,
-                'toadd'           => $toadd
-            ]
-        );
-        if ($ID) {
-            $ticket_duration = ProjectTask_Ticket::getTicketsTotalActionTime($this->getID());
-            echo "<br>";
-            printf(
-                __('%1$s: %2$s'),
-                __('Tickets duration'),
-                Html::timestampToString($ticket_duration, false)
-            );
-            echo '<br>';
-            printf(
-                __('%1$s: %2$s'),
-                __('Total duration'),
-                Html::timestampToString(
-                    $ticket_duration + $this->fields["effective_duration"],
-                    false
-                )
-            );
-        }
-        echo "</td></tr>\n";
-
-        echo "<tr class='tab_bg_1'>";
-        echo "<td>" . __('Description') . "</td>";
-        echo "<td colspan='3'>";
-        Html::textarea([
-            'name'            => 'content',
-            'enable_richtext' => true,
-            'editor_id'       => "description$rand_description",
-            'value'           => RichText::getSafeHtml($this->fields["content"], true),
+            $duration_dropdown_to_add[] = $i * HOUR_TIMESTAMP;
+        }
+
+        $this->initForm($ID, $options);
+        TemplateRenderer::getInstance()->display('pages/tools/project_task.html.twig', [
+            'id'                       => $ID,
+            'item'                     => $this,
+            'params'                   => $options,
+            'parent'                   => Project::getById($projects_id),
+            'projects_id'              => $projects_id,
+            'projecttasks_id'          => $projecttasks_id,
+            'recursive'                => $recursive,
+            'duration_dropdown_to_add' => $duration_dropdown_to_add,
+            'ticket_duration'          => $ticket_duration,
+            'total_duration'           => $total_duration,
+            'safe_content'             => RichText::getSafeHtml($this->fields["content"], true),
```

Comment:
This should not be needed. Indeed, `|safe_html|escape` filters are already applied by textarea field macro.

```suggestion
            'content'             => $this->fields["content"],
```